salt/utils/win_functions.py

```diff
@@ -214,10 +214,24 @@ def escape_meta_chars(m):
     return meta_re.sub(escape_meta_chars, arg)
 
 
-def refresh_environment():
+def broadcast_setting_change(setting='Environment'):
     '''
-    Send a WM_SETTINGCHANGE Broadcast to Windows to refresh the Environment
-    variables for new processes.
+    Send a WM_SETTINGCHANGE Broadcast to all Windows
+
+    Args:
+
+        setting (str):
+            A string value representing the portion of the system that has been
+            updated and needs to be refreshed. Default is ``Environment``. These
+            are some common values:
+
+            - "Environment" : to effect a change in the environment variables
+            - "intl" : to effect a change in locale settings
+            - "Policy" : to effect a change in Group Policy Settings
+            - a leaf node in the registry
+            - the name of a section in the ``Win.ini`` file
+
+            `See here <https://msdn.microsoft.com/en-us/library/ms725497%28VS.85%29.aspx>`
```

Comment:
Suggested Changes:
def broadcast_setting_change(message='Environment')

Sends message to all top-level windows, only those application windows listing for the particular message sent will react.  Check with application developers for a particular application, however the following is a list of common broadcast messages some application windows may listen for:

See  lParam witin .... for sending and   GWL_WNDPROC within https://msdn.microsoft.com/en-us/library/windows/desktop/ms633591(v=vs.85).aspx for receiving.

May include this as a comment in the code
Python listener would involve old_function = win32gui.SetWindowLong(window_handle,win32con.GWL_WNDPROC, new_function
To give people a glue if they are trying to do something like this.

End of Suggestions.

It seems its just a message which is sent, and the expectation is something is listening for it.  The string could be anything and the app could be listening for anything.    You could even send "salt-minion" and have the salt-minion listen for it and do a high-state.

Reply:
Changed the name of the parameter to message. I left the default as "Environment". I believe that to be a reasonable default.

Reply:
Update the text place I think it explains it better.  Sends ...

Reply:
Ping @rallytime